app/controllers/curation_concerns/generic_works_controller.rb

```diff
@@ -30,6 +31,14 @@ def identifiers
     end
   end
  
+  def assign_visibility
+    byebug
+    if (params["generic_work"]["visibility"]=="true")
+      params['generic_work']['visibility'] = Hydra::AccessControls::AccessRight::VISIBILITY_TEXT_VALUE_PRIVATE
+    else  
+      params['generic_work']['visibility'] = 'open'
```

Comment:
Using the hydra access controls constant as you have above is preferable to the hard coded string 'open', so this can be the appropriate Hydra::AccessControls::AccessRight constant as well.  I recall the constant evaluates to 'open', but the way you have it above seems more appropriate.